File: src/com/rian/difficultycalculator/attributes/PerformanceAttributes.java
package com.rian.difficultycalculator.attributes;

/**
 * A structure containing the performance values of a score.
 */
public class PerformanceAttributes {
    /**
     * Calculated score performance points.
     */
    public double total;

    /**
     * The aim performance value.
     */
    public double aim;

    /**
     * The speed performance value.
     */
    public double speed;

    /**
     * The accuracy performance value.
     */
    public double accuracy;

    /**
     * The flashlight performance value.
     */
    public double flashlight;

    /**
     * The amount of misses that are filtered out from slider breaks.
     */
    public double effectiveMissCount;
}

File: src/com/rian/osu/difficulty/attributes/PerformanceAttributes.kt
package com.rian.osu.difficulty.attributes

/**
 * A structure containing the performance values of a score.
 */
abstract class PerformanceAttributes {
    /**
     * Calculated score performance points.
     */
    @JvmField
    var total = 0.0

    /**
     * The aim performance value.
     */
    @JvmField
    var aim = 0.0

    /**
     * The accuracy performance value.
     */
    @JvmField
    var accuracy = 0.0

    /**
     * The flashlight performance value.
     */
    @JvmField
    var flashlight = 0.0

    /**
     * The amount of misses including slider breaks.
     */
    @JvmField
    var effectiveMissCount = 0.0
}
